staging/src/k8s.io/apiserver/pkg/apis/audit/validation/validation.go

```diff
@@ -60,3 +65,38 @@ func validateLevel(level audit.Level, fldPath *field.Path) field.ErrorList {
 		return field.ErrorList{field.NotSupported(fldPath, level, validLevels)}
 	}
 }
+
+func validateNonResourceURLs(urls []string, fldPath *field.Path) field.ErrorList {
+	var allErrs field.ErrorList
+	for _, url := range urls {
+		if url == "*" {
+			continue
+		}
+
+		if !strings.HasPrefix(url, "/") {
+			allErrs = append(allErrs, field.Invalid(fldPath, url, "invalid non-resource URL format"))
+		}
+
+		count := strings.Count(url, "*")
+		if count > 1 || (count == 1 && !strings.HasSuffix(url, "*")) {
+			allErrs = append(allErrs, field.Invalid(fldPath, url, "invalid non-resource URL format, only support one tailing * at the end"))
+		}
+	}
+	return allErrs
+}
+
+func validateResources(groupResources []audit.GroupResources, fldPath *field.Path) field.ErrorList {
+	var allErrs field.ErrorList
+	for _, groupResource := range groupResources {
+		if len(groupResource.Group) == 0 {
+			continue
+		}
+
+		// an error is returned for group name like rbac.authorization.k8s.io/v1beta1
+		// rbac.authorization.k8s.io is the valid one
+		if msgs := genericvalidation.NameIsDNSSubdomain(groupResource.Group, false); len(msgs) != 0 {
```

Comment:
I don't know if this is the correct way to validate a group name. There should be something to parse a group version such as:

https://godoc.org/k8s.io/apimachinery/pkg/runtime/schema#ParseGroupResource
https://godoc.org/k8s.io/apimachinery/pkg/runtime/schema#ParseGroupVersion

cc @timstclair @sttts any comments here?